src/fides/api/ops/schemas/saas/connector_template.py

```diff
@@ -1,28 +1,43 @@
-import yaml
+from typing import Optional
+
 from fideslang.models import Dataset
 from pydantic import BaseModel, validator
 
 from fides.api.ops.schemas.saas.saas_config import SaaSConfig
+from fides.api.ops.util.saas_util import (
+    load_config_from_string,
+    load_dataset_from_string,
+)
 
 
 class ConnectorTemplate(BaseModel):
     """
-    A collection of artifacts that make up a complete SaaS connector (SaaS config, dataset, etc.)
+    A collection of artifacts that make up a complete
+    SaaS connector (SaaS config, dataset, icon, functions, etc.)
     """
 
     config: str
     dataset: str
-    icon: str
+    icon: Optional[str]
+    functions: Optional[str]
     human_readable: str
 
     @validator("config")
     def validate_config(cls, config: str) -> str:
         """Validates the config at the given path"""
-        SaaSConfig(**yaml.safe_load(config).get("saas_config"))
+        saas_config = SaaSConfig(**load_config_from_string(config))
+        if saas_config.fides_key != "<instance_fides_key>":
+            raise ValueError(
+                "Hard-coded fides_key detected in the config, replace all instances of it with <instance_fides_key>"
```

Comment:
Checking that the `fides_key` is set to `<instance_fides_key>` so we know where to plug in the customer assigned fides key